definitions/push-application-snapshot/push-application-snapshot.yaml

```
@@ -0,0 +1,71 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: push-application-snapshot-task
+spec:
+  params:
+    - name: mappedApplicationSnapshot
+      description: |
+        JSON string representing the ApplicationSnapshot
+        For example:
+        """
+        {
+          "application": "foo",
+          "images": [
+            {
+              "component": "component_fizz_1",
+              "pullspec": "quay.io/src_namespace/src_repo_name@sha256:digest",
+              "targetRepo": "quay.io/dest_namespace/dest_repo_name"
+            },
+            {
+              "component": "component_bang_3",
+              "pullspec": "quay.io/src_namespace/src_repo_name",
+              "targetRepo": "quay.io/dest_namespace/dest_repo_name"
+            }
+          ]
+        }
+        """
+        Note:
+        * The pullspec digest (e.g. @sha256:...) may be ommited.
+        * If the pullspec digest is ommited, then assume "latest".
+      type: string
+    - name: retries
+      description: Retry copy N times.
+      type: int
+      default: 0
+    - name: destinationTag
+      description: option tag to give destination images.
+      type: string
+      default: ""
+  steps:
+    - name: mappedApplicationSnapshot
+      image: http://quay.io/hacbs-release/release-utils:latest
+      script: |
+        if [[ -z "$(params.mappedApplicationSnapshot)" ]]
+        then
+          exit 1 # Missing URL parameters
+        fi
+        app=$(jq -r '.application' <<<$(params.mappedApplicationSnapshot))
+        tag="$(params.destinationTag)"
+        printf 'Begining push task for application: "%s"\n' "$app"
+        while read line;
+        do
+          set -- $(jq -cr '.[]' <<<"$line" | xargs)
+          s_Digest=$(skopeo inspect --format '{{.Digest}}' "$2" 2>/dev/null)
```

```suggestion
          source_digest=$(skopeo inspect --format '{{.Digest}}' "$2" 2>/dev/null)
```

Note: I didn't know about `--format`. That's nice!